### PR TITLE
[VIDCS-922] Video Transformers sample

### PR DESCRIPTION
**Pull request by goncalocostamendes**
https://jira.vonage.com/browse/VIDCS-922

### CODE REVIEW

**Comment by IGitGotIt** [comment]
Can we have the branch name be the JIRA ticket number - PLEASE PLEASE...

File: Video-Transformers/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+OpenTok developer center](https://tokbox.com/developer/guides/vonage-media-processor/ios).
+
+You can use pre-built transformers in the Vonage Media Processor library or create your own custom video transformer to apply to published video.
+
+You can use the OTPublisherKit.videoTransformers properties to apply video transformers to a stream.
+
+You can combine the Vonage Media library transformer with custom transformer or apply multiple custom transformers by adding multiple PublisherKit.VideoTransformer objects to the ArrayList used for the OTPublisherKit.videoTransformers property.
+
+To clear video transformers for a publisher, set the OTPublisherKit.videoTransformers property to an empty array.
+
```
Comment (IGitGotIt) [inline Video-Transformers/README.md:15]: Some sample code will be nice

File: Broadcast-Ext/OpenTok Live/OTBroadcastExtAudioDevice.m [hunk 2]
```diff
@@ -47,6 +47,8 @@
 #endif
 
 static double kPreferredIOBufferDuration = 0.01;
+static double kToMicroSecond = 1000000;
```
Comment (IGitGotIt) [inline Broadcast-Ext/OpenTok Live/OTBroadcastExtAudioDevice.m:50]: Why is the Broadcast Ext in this PR. Can we create a separate PR/ticket. Thanks

File: Custom-Audio-Driver/External-Audio-Device/OTDefaultAudioDevice.m [hunk 3]
```diff
@@ -47,6 +47,8 @@
 #endif
 
 static double kPreferredIOBufferDuration = 0.01;
+static double kToMicroSecond = 1000000;
```
Comment (IGitGotIt) [inline Custom-Audio-Driver/External-Audio-Device/OTDefaultAudioDevice.m:50]: The ReadMe says it is build on top of the Basic App, but this is a custom audio driver code.

File: Video-Transformers/Podfile [hunk 4]
```diff
@@ -0,0 +1,8 @@
+require_relative '../OpenTokSDKVersion'
+
+platform :ios, '13'
+use_frameworks!
+
+target 'Video-Transformers' do
+  pod 'VonageClientSDKVideo', '2.25.2'
```
Comment (IGitGotIt) [inline Video-Transformers/Podfile:7]: Is the pod name ok.. I am not seeing it on cocoapods. also if correct I hope it is XC framework.

File: Video-Transformers/Video-Transformers/ViewController.m [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+    if (error)
+    {
+        [self showAlert:[error localizedDescription]];
+    }
+    
+    [self.view addSubview:_publisher.view];
+    [_publisher.view setFrame:CGRectMake(0, 0, widgetWidth, widgetHeight)];
+
+    
+    NSString *resourcePath = [[NSBundle mainBundle] pathForResource:@"selfie_segmentation" ofType:@"tflite" inDirectory:@"OpenTok.bundle"];
```
Comment (IGitGotIt) [inline Video-Transformers/Video-Transformers/ViewController.m:144]: This should be done once.. in init maybe?

File: Video-Transformers/README.md (hunk 1 above)
Comment (goncalocostamendes) [reply Video-Transformers/README.md:15]: I will add it

File: Broadcast-Ext/OpenTok Live/OTBroadcastExtAudioDevice.m (hunk 2 above)
Comment (goncalocostamendes) [reply Broadcast-Ext/OpenTok Live/OTBroadcastExtAudioDevice.m:50]: I am removing those changes

File: Custom-Audio-Driver/External-Audio-Device/OTDefaultAudioDevice.m (hunk 3 above)
Comment (goncalocostamendes) [reply Custom-Audio-Driver/External-Audio-Device/OTDefaultAudioDevice.m:50]: This is  on a different sample. I will remove those changes

File: Video-Transformers/Podfile (hunk 4 above)
Comment (goncalocostamendes) [reply Video-Transformers/Podfile:7]: @chetanvangadiTokbox any idea why VonageClientSDKVideo does not appear on cocoapods? It works well running pod install, but it seems it is not present visually on the website

File: Video-Transformers/Video-Transformers/ViewController.m (hunk 5 above)
Comment (goncalocostamendes) [reply Video-Transformers/Video-Transformers/ViewController.m:144]: yes, we can do that

File: Video-Transformers/Video-Transformers/ViewController.m [hunk 6]
```diff
@@ -0,0 +1,340 @@
...
+    if (resourcePath) {
+        // File found, use it as needed
+        // For example, you can read the contents of the file:
+        NSError *error = nil;
+        NSString *contents = [NSString stringWithContentsOfFile:resourcePath encoding:NSUTF8StringEncoding error:&error];
+        if (contents) {
+            NSLog(@"%@", contents);
```
[the remark this answers is not in the document]
Comment (goncalocostamendes) [reply Video-Transformers/Video-Transformers/ViewController.m:151]: sorry, this should not be present. Was only to validate if the tflite model was being added to the bundle. We don't need it

**Comment by goncalocostamendes** [comment]
@IGitGotIt PR replaced by https://github.com/opentok/opentok-ios-sdk-samples/pull/289

**Pull request closed without merging**